sky/clouds/kubernetes.py

```diff
@@ -319,10 +322,14 @@ def make_deploy_resources_variables(
             zones: Optional[List['clouds.Zone']],
             dryrun: bool = False) -> Dict[str, Optional[str]]:
         del cluster_name, zones, dryrun  # Unused.
+        meta_region = self.DEFAULT_IMAGE_META_REGION
         if region is None:
             context = kubernetes_utils.get_current_kube_config_context_name()
         else:
             context = region.name
+            img_meta_region = region.name.split('-')[0]
+            if img_meta_region in self.IMAGE_META_REGIONS:
+                meta_region = img_meta_region
```

Comment:
Actually, the region name is just the context name of the kubernetes cluster, which won't contain the location information, so probably we have to always use us.